# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/vm/NativeObject-inl.h
# Commit: 8129ee5398d5
# Full Hash: 8129ee5398d5ce34183dd2ca51c3aa1d27e9c318
# Author: Doug Thayer <dothayer@mozilla.com>
# Date: 2023-01-20 03:46:23
# Regressor Bug: 1809359
# File Overlap Count: 1
# Description:
#   Bug 1809359 - Inline SetElementMegamorphic r=jandem
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D166401
# ==============================================================================

diff -r a4156a9d0b2b -r 8129ee5398d5 js/src/vm/NativeObject-inl.h
--- a/js/src/vm/NativeObject-inl.h	Thu Jan 19 22:22:32 2023 +0000
+++ b/js/src/vm/NativeObject-inl.h	Thu Jan 19 23:53:40 2023 +0000
@@ -843,19 +843,21 @@
 
 // Like AddDataProperty but optimized for plain objects. Plain objects don't
 // have an addProperty hook.
-MOZ_ALWAYS_INLINE bool AddDataPropertyToPlainObject(JSContext* cx,
-                                                    Handle<PlainObject*> obj,
-                                                    HandleId id,
-                                                    HandleValue v) {
+MOZ_ALWAYS_INLINE bool AddDataPropertyToPlainObject(
+    JSContext* cx, Handle<PlainObject*> obj, HandleId id, HandleValue v,
+    uint32_t* resultSlot = nullptr) {
   MOZ_ASSERT(!id.isInt());
 
   uint32_t slot;
-  if (!NativeObject::addProperty(cx, obj, id,
-                                 PropertyFlags::defaultDataPropFlags, &slot)) {
+  if (!resultSlot) {
+    resultSlot = &slot;
+  }
+  if (!NativeObject::addProperty(
+          cx, obj, id, PropertyFlags::defaultDataPropFlags, resultSlot)) {
     return false;
   }
 
-  obj->initSlot(slot, v);
+  obj->initSlot(*resultSlot, v);
 
   MOZ_ASSERT(!obj->getClass()->getAddProperty());
   return true;